test/common/http/http2/frame_replay_test.cc

```diff
@@ -145,13 +162,24 @@ TEST_F(ResponseFrameCommentTest, SimpleExamplePlain) {
 
   // Validate HEADERS decode.
   ClientCodecFrameInjector codec;
-  codec.write(WellKnownFrames::defaultSettingsFrame());
-  codec.write(WellKnownFrames::initialWindowUpdateFrame());
+  TestClientConnectionImpl connection(codec.client_connection_, codec.client_callbacks_,
+                                      codec.stats_store_, codec.settings_,
+                                      Http::DEFAULT_MAX_REQUEST_HEADERS_KB);
+  // Create a new stream.
+  codec.request_encoder_ = &connection.newStream(codec.response_decoder_);
+  codec.request_encoder_->getStream().addCallbacks(codec.client_stream_callbacks_);
+  // Setup a single stream to inject frames as a reply to.
+  TestHeaderMapImpl request_headers;
+  HttpTestUtility::addDefaultHeaders(request_headers);
+  codec.request_encoder_->encodeHeaders(request_headers, true);
```

Comment:
Can you factor out this boiler plate into a standalone function in this `_test.cc`? It would make the test logic easier to read.